workitem.go

```diff
@@ -135,6 +135,56 @@ func (c *WorkitemController) Update(ctx *app.UpdateWorkitemContext) error {
 	})
 }
 
+// Reorder reorders the workitem
+func (c *WorkitemController) Reorder(ctx *app.ReorderWorkitemContext) error {
+	return application.Transactional(c.db, func(appl application.Application) error {
+
+		if ctx.Payload == nil || ctx.Payload.Data == nil || ctx.Payload.Data.ID == nil {
+			jerrors, _ := jsonapi.ErrorToJSONAPIErrors(errors.NewBadParameterError("data.id", nil))
+			return ctx.NotFound(jerrors)
+		}
+
+		wi, err := appl.WorkItems().Load(ctx, *ctx.Payload.Data.ID)
+		if err != nil {
+			jerrors, _ := jsonapi.ErrorToJSONAPIErrors(goa.ErrNotFound(fmt.Sprintf("Error reordering work item: %s", err.Error())))
+			return ctx.NotFound(jerrors)
+		}
+		err = ConvertJSONAPIToWorkItem(appl, *ctx.Payload.Data, wi)
+		if err != nil {
+			jerrors, _ := jsonapi.ErrorToJSONAPIErrors(goa.ErrBadRequest(fmt.Sprintf("Error reordering work item: %s", err.Error())))
+			return ctx.BadRequest(jerrors)
+		}
+		wi, err = appl.WorkItems().Reorder(ctx, *wi)
+		if err != nil {
+			switch err := err.(type) {
+			case errors.BadParameterError:
+				jerrors, _ := jsonapi.ErrorToJSONAPIErrors(goa.ErrBadRequest(fmt.Sprintf("Error reordering work item: %s", err.Error())))
+				return ctx.BadRequest(jerrors)
+			case errors.NotFoundError:
+				jerrors, _ := jsonapi.ErrorToJSONAPIErrors(goa.ErrNotFound(err.Error()))
+				return ctx.NotFound(jerrors)
+			case errors.VersionConflictError:
+				jerrors, _ := jsonapi.ErrorToJSONAPIErrors(goa.ErrBadRequest(fmt.Sprintf("Error reordering work item: %s", err.Error())))
+				return ctx.BadRequest(jerrors)
+			default:
+				log.Printf("Error reordering work items: %s", err.Error())
+				jerrors, _ := jsonapi.ErrorToJSONAPIErrors(goa.ErrInternal(err.Error()))
+				return ctx.InternalServerError(jerrors)
+			}
+		}
+
+		wi2 := ConvertWorkItem(ctx.RequestData, wi)
+		resp := &app.WorkItem2Single{
+			Data: wi2,
+			Links: &app.WorkItemLinks{
+				Self: buildAbsoluteURL(ctx.RequestData),
+			},
+		}
+
+		return ctx.OK(resp)
```

Comment:
@pranavgore09 ctx.OK() sends the HTTP respose **with** status code and not **for** status code